### ISSUE TITLE
[Q]Is there a way to log different metrics with its own step setting?

### ISSUE THREAD

**Issue by lhq12** [wandb/wandb#5789]
I want to use wandb to track different metrics with differently grained step. For example, I use `wandb.log({'train/pseudo-label-accuracy': acc}, step=epoch*num_batches + batch)` to track train/pseudo-label-accuracy of each batch and use` wandb.log({'test/accuracy': acc}, step=epoch)` to track test/accuracy of each epoch. But I found `test/accuracy` was dropped. How can I fix this?

**Comment by lhq12** [comment]
And there is another scenario, where the model is first pre-trained and then fine-tuned. At pre-trianing stage, a metric will be logged for 400 steps . Then at fine-tuning stage, the metric will be dropped if I log it start from step 0.

**Comment by thanos-wandb** [comment]
Hi @lhq12 thanks for writing in! You could have a different step using `define_metric`. Please check our [documentation here](https://docs.wandb.ai/guides/track/log/customize-logging-axes) and the link to a Colab example.  Would this work for your use case?

**Comment by lhq12** [comment]
Hi @thanos-wandb, thank you for the quick reply and it works

**Comment by thanos-wandb** [comment]
Perfect, thanks for confirming @lhq12 ! I will now close this ticket, and please free free to reach out here if you had any other questions/issues.
